src/watertap_contrib/seto/costing/units/chemical_softening_zo.py

```diff
@@ -0,0 +1,968 @@
+    # Admin cost
+    blk.admin_op_cost_constraint = Constraint(
+        expr=blk.admin_op_cost
+        == (
+            chem_soft.admin_op_coeff
+            * (
+                (
+                    pyunits.convert(
+                        blk.unit_model.properties_in[0].flow_vol,
+                        to_units=pyunits.megagallon / pyunits.day,
+                    )
+                )
+                ** chem_soft.admin_op_exp
+            )
+        )
+    )
+
+    # Sum of all capital costs
+    blk.capital_cost_constraint = Constraint(
+        expr=blk.capital_cost
+        == blk.mix_tank_capital_cost
+        + blk.floc_tank_capital_cost
+        + blk.sed_basin_capital_cost
+        + blk.recarb_basin_capital_cost
+        + blk.recarb_basin_source_capital_cost
+        + blk.lime_feed_system_capital_cost
+        + blk.admin_capital_cost
+    )
+
+    # Sum of all operational costs
+
+    if blk.unit_model.config.silica_removal:
+        blk.fixed_operating_cost_constraint = Constraint(
+            expr=blk.fixed_operating_cost
+            == blk.mix_tank_op_cost
+            + blk.floc_tank_op_cost
+            + blk.sed_basin_op_cost
+            + blk.recarb_basin_op_cost
+            + blk.lime_feed_op_cost
+            + blk.lime_sludge_mngt_op_cost
+            + blk.admin_op_cost
+        )
+    else:
+        blk.fixed_operating_cost_constraint = Constraint(
+            expr=blk.fixed_operating_cost
+            == blk.mix_tank_op_cost
+            + blk.floc_tank_op_cost
+            + blk.sed_basin_op_cost
+            + blk.recarb_basin_op_cost
+            + blk.lime_feed_op_cost
+            + blk.lime_sludge_mngt_op_cost
+            + blk.admin_op_cost
+        )
+
+    blk.mixer_power_constraint = Constraint(
+        expr=blk.mixer_power
+        == (blk.unit_model.vel_gradient_mix**2)
+        * blk.unit_model.volume_mixer
+        * blk.unit_model.properties_in[0].params.visc_d
+    )
+
+    blk.floc_power_constraint = Constraint(
+        expr=blk.floc_power
+        == (blk.unit_model.vel_gradient_floc**2)
+        * blk.unit_model.volume_floc
+        * blk.unit_model.properties_in[0].params.visc_d
+    )
+
+    blk.electricity_flow_constraint = Constraint(
+        expr=blk.electricity_flow
+        == pyunits.convert((blk.mixer_power + blk.floc_power), to_units=pyunits.kW)
+    )
+
+    CaO_dosing = pyunits.convert(
+        blk.unit_model.CaO_dosing
+        + pyunits.convert(
+            blk.unit_model.excess_CaO * blk.unit_model.properties_in[0].flow_vol,
+            to_units=pyunits.kg / pyunits.day,
+        ),
+        to_units=pyunits.kg / pyunits.year,
+    )
+
+    mgcl2_dosing = pyunits.convert(
+        blk.unit_model.MgCl2_dosing * blk.unit_model.properties_in[0].flow_vol,
+        to_units=pyunits.kg / pyunits.year,
+    )
+
+    co2_dosing = pyunits.convert(
+        blk.unit_model.CO2_first_basin + blk.unit_model.CO2_second_basin,
+        to_units=pyunits.kg / pyunits.year,
+    )
```

Comment:
Any reason why these shouldn't be declared as Expressions on the model for later access, as opposed to local aliases?

Reply:
Updated these to be expressions.

Reply:
It is fine either way I think. The values are reportable via the ``aggregate_flow_costs`` or whatever the aggregated variable is. Thought it can be nice to directly access the costed flows via the unit model's costing block. (Also, I imagine this gives a WARNING since they are Expressions costing flows and aren't Constraints, but I don't think that change is worth making, especially for this PR).